src/spatial_reaction_systems/spatial_ODE_systems.jl

```diff
@@ -0,0 +1,182 @@
+    end
+end
+
+### ODEProblem ###
+
+# Creates an ODEProblem from a LatticeReactionSystem.
+function DiffEqBase.ODEProblem(lrs::LatticeReactionSystem, u0_in, tspan,
+                               p_in = DiffEqBase.NullParameters(), args...;
+                               jac = true, sparse = jac, kwargs...)
+    is_transport_system(lrs) || error("Currently lattice ODE simulations only supported when all spatial reactions are transport reactions.")
+    
+    # Converts potential symmaps to varmaps.
+    u0_in = symmap_to_varmap(lrs, u0_in)
+    p_in = (p_in isa Tuple{<:Any,<:Any}) ? (symmap_to_varmap(lrs, p_in[1]),symmap_to_varmap(lrs, p_in[2])) : symmap_to_varmap(lrs, p_in)    # Parameters can be given in Tuple form (where the first element is the vertex parameters and the second the edge parameters). In this case, we have to covert each element separately.
+
+    # Converts u0 and p to their internal forms.
+    u0 = lattice_process_u0(u0_in, species(lrs), lrs.num_verts)                                     # u0 becomes a vector ([species 1 at vertex 1, species 2 at vertex 1, ..., species 1 at vertex 2, ...]).
+    vert_ps, edge_ps = lattice_process_p(p_in, vertex_parameters(lrs), edge_parameters(lrs), lrs)   # Both vert_ps and edge_ps becomes vectors of vectors. Each have 1 element for each parameter. These elements are length 1 vectors (if the parameter is uniform), or length num_verts/nE, with unique values for each vertex/edge (for vert_ps/edge_ps, respectively).
+
+    # Creates ODEProblem.
+    ofun = build_odefunction(lrs, vert_ps, edge_ps, jac, sparse)        # Builds the ODEFunction.
+    return ODEProblem(ofun, u0, tspan, vert_ps, args...; kwargs...)     # Creates a normal ODEProblem.
+end
+
+# Builds an ODEFunction for a spatial ODEProblem.
+function build_odefunction(lrs::LatticeReactionSystem, vert_ps::Vector{Vector{T}},
+                           edge_ps::Vector{Vector{T}}, use_jac::Bool, sparse::Bool) where {T}
+    # Prepares (non-spatial) ODE functions and list of spatially moving species and their rates.
+    ofunc = ODEFunction(convert(ODESystem, lrs.rs); jac = use_jac, sparse = false)                  # Creates the (non-spatial) ODEFunction corresponding to the (non-spatial) reaction network.
+    ofunc_sparse = ODEFunction(convert(ODESystem, lrs.rs); jac = use_jac, sparse = true)            # Creates the same function, but sparse. Could insert so this is only computed for sparse cases.
+    transport_rates_speciesmap = compute_all_transport_rates(vert_ps, edge_ps, lrs)                 # Creates a map (Vector{Pair}), mapping each species that is transported to a vector with its transportation rate. If the rate is uniform across all edges, the vector will be length 1 (with this value), else there will be a separate value for each edge.
+    transport_rates = Pair{Int64, Vector{T}}[findfirst(isequal(spat_rates[1]), species(lrs)) => spat_rates[2]
```

Comment:
Why not use `speciesmap` or make a lookup dictionary to build a mapping from species to index? That should be faster for big systems than searching like this for every species.

Reply:
Yes, that is much better, thanks for pointing it out.

There are all these convenience functions, and I can never keep track of which are defined in Catalyst, which in MTK, which can only be accessed as a field (if there are any like that these days), and which doesn't exist. 

At some point I will try to create a list (maybe when we write the dev-docs, seems like a good resource to have(

Reply:
The Catalyst API lists those we define.